### PR TITLE
feat(jest): create jest config package for unit testing

### PR DESCRIPTION
**Pull request by SychO9**
**Changes proposed in this pull request:**
This is part one of two PRs to introduce tests to our JavaScript codebase. This PR lays the foundation by introducing a `flarum-jest-config` package and the necessary changes to begin unit testing. An example core `abbreviateNumber` util test is added as well.

**Reviewers should focus on:**
* Anything I might be missing about introducing Jest testing.

Part two is here: #3679

**Necessity**

- [x] Has the problem that is being solved here been clearly explained?
- [x] If applicable, have various options for solving this problem been considered?
- [x] For core PRs, does this need to be in core, or could it be in an extension?
- [x] Are we willing to maintain this for years / potentially forever?

**Confirmed**

- [x] Frontend changes: tested on a local Flarum installation.
- [x] Backend changes: tests are green (run `composer test`).
- [ ] Core developer confirmed locally this works as intended.
- [x] Tests have been added, or are not appropriate here.

**Required changes:**

- [ ] Related documentation PR: (Remove if irrelevant)
- [ ] Related core extension PRs: (Remove if irrelevant)

### CODE REVIEW

**Comment by SychO9** [comment]
need to add the workflow

File: js-packages/jest-config/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Jest config for Flarum
+
+This package generates a [Jest](https://jestjs.io/) config object to run unit & integration tests on Flarum extensions.
```
Comment (askvortsov1) [inline js-packages/jest-config/README.md]:
```suggestion
This package provides a [Jest](https://jestjs.io/) config object to run unit & integration tests on Flarum extensions.
```

File: js-packages/webpack-config/index.js [hunk 2]
```diff
@@ -72,33 +72,11 @@ module.exports = function (options = {}) {
         {
           // Matches .js, .jsx, .ts, .tsx
           // See: https://regexr.com/5snjd
-          test: /\.(j|t)sx?$/,
+          test: /\.[jt]sx?$/,
```
Comment (askvortsov1) [inline js-packages/webpack-config/index.js:75]: A lot cleaner, yay!

**Review by askvortsov1 — APPROVED** [review]
Wonderful, thanks so much for making this work! I think the only uncertainty I have is whether it would be better to put test code next to the source code, either in the same folder or in a `__tests__` subdir. That way, it's easier to keep track of source code having corresponding tests, although coverage should help with that too.

**Comment by SychO9** [comment]
> Wonderful, thanks so much for making this work! I think the only uncertainty I have is whether it would be better to put test code next to the source code, either in the same folder or in a __tests__ subdir. That way, it's easier to keep track of source code having corresponding tests, although coverage should help with that too.

I don't have any strong opinions about this at the moment. I'm okay with it either way. We will probably find out as we write more tests in the future.

**Comment by askvortsov1** [comment]
Let's keep as is for now then, LGTM!

**Pull request merged**